cmd/docker/docker_test.go

```diff
@@ -55,3 +58,51 @@ func TestVersion(t *testing.T) {
 	assert.NilError(t, err)
 	assert.Check(t, is.Contains(b.String(), "Docker version"))
 }
+
+func TestErrorCodePluginBad(t *testing.T) {
+	cli, err := command.NewDockerCli(command.WithInputStream(discard), command.WithCombinedStreams(ioutil.Discard))
+	assert.NilError(t, err)
+	cmd := &cobra.Command{
+		Use:   "test",
+		Short: "Test Plugin command that exits",
+		RunE: func(_ *cobra.Command, _ []string) error {
+			os.Exit(2)
+			return nil
+		},
+	}
+
+	if os.Getenv("CHECK_RETURN_CODE") == "1" {
+		executeCmd(cli, cmd)
+		return
+	}
```

Comment:
This reexecing with an envvar is a bit convoluted, I'd suggest to make it an e2e test instead. Putting it in `e2e/cli-plugins/run_test.go` would result it in being alongside a bunch of similar tests.

To enable that you could extend `cli-plugins/examples/helloworld/` to have a subcommand which exits with an error, or add a new plugin for e2e testing under `e2e/cli-plugins/plugins` alongside the `badmeta` one.

Reply:
Perfect I was looking for something like this I didn't know it was in place